src/util/collections.ts

```diff
@@ -0,0 +1,26 @@
+export function uniqueArray<V>(original: Array<V> | null = null): Array<V> {
+  if (original) {
+    let cloned = new Array<V>(original.length);
+    for (let i = 0, k = original.length; i < k; ++i) cloned[i] = original[i];
+    return cloned;
+  }
+  return new Array<V>();
```

Comment:
Fastest and simplest unique for array which I know is:
```js
function unique<T>(arr: Array<T>): Array<T> {
  return Array.from(new Set<T>(arr));
}
```
But it require iterators. And may be not perfect align to current stale of AS
```

Reply:
Ahh, I got what is that) May be better rename this to `makeArray` and `makeSet` for avoid confusion?

Reply:
Yeah, the code currently assumes that we'd at some point try to compile the compiler with asc, with some features not yet supported.

Reply:
I got it. But probably make sense rename it to `make*` instead `unique*` ?

Reply:
Fine with `make*` as well, going to change :)